benchmark/README.rst

```diff
@@ -0,0 +1,53 @@
+Benchmarking tool for PennyLane
+===============================
+
+``benchmark.py`` is a simple benchmarking script that can do timings, performance plots and
+basic profiling. The benchmarks can be run on different PL devices, and the results compared.
+
+The script provides a uniform way of evaluating PL code performance, and the way it changes
+between commits. It is meant to be run on a Git repository containing the PL code, and will
+automatically extract the short hash of the currently checked out commit to label the results.
+
+The script has three main functions.
+
+
+* Profiling::
+
+      python3 benchmark.py profile bm_mutable_rotations
+      bash view_pstats.sh pennylane_*.pstats
+
+  The profiling stats are saved in a .pstats file, which can be viewed using various tools.
+  For example, the ``view_pstats.sh`` BASH script uses ``gprof2dot`` to present the data as a
+  call graph.
+
+* Timing::
+
+      python3 benchmark.py -d default.qubit,default.tensor time bm_mutable_rotations
```

Comment:
Should we suppress the typical annoying tensorflow warnings when running benchmarks?

Reply:
Probably, if they're not important. How does one do that?

Reply:
there's a special environment variable for doing it in tensorflow (at least there was in older versions), but I was thinking of displaying only certain log levels would be more general?

Reply:
> * can we print key info in colour? e.g., the Timing per loop in the timing benchmark (helps find key info when lots of TF warning messages are also showing

I added some colors :)

>   * when I ran with the plot arg, I saw two progress bars in the terminal (even when I just had one device), but what they were indicating the way they were updating was confusing (kept going back to 0%)

One shows the progress in the kernels, the other within the current kernel.